src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2973,6 +2979,26 @@ private string GetAssemblyPathInGac(AssemblyNameExtension assemblyName, SystemPr
         /// <returns>True if there was success.</returns>
         public override bool Execute()
         {
+            if (UseResolveAssemblyReferenceService && BuildEngine is IRarBuildEngine rarBuildEngine)
+            {
+                using var client = new RarClient(rarBuildEngine);
+
+                var connected = client.Connect();
+                if (!connected)
+                {
+                    if (client.CreateNode())
+                    {
+                        connected = client.Connect(5000);
```

Comment:
Might want to comment—why 5000?

Reply:
Wanted to give enough time for node to startup, since `CreateNode` only starts process and don't wait for the connection. For example: OutOfProc has timeout for new connection 30 seconds.